### PR TITLE
Add CentOS jobs

### PR DESCRIPTION
**Pull request by cottsay**
This change adds all of the standard jobs on CentOS 7.

Integrating the Dockerfiles at this point would be a huge effort, so I decided it was easier to start a separate Dockerfile from scratch. The rest of the infrastructure is used as-is.

There are lots of compiler warnings and test failures, but this is expected at this point. Hopefully having these jobs will make it easier to drive those numbers down.

Since the RHEL/CentOS work is still in early stages, it might be a good idea to exclude these jobs from the trigger jobs. I'm hoping for some feedback on that.

### CODE REVIEW

**Comment by cottsay** [comment]
Example CI job: https://ci.ros2.org/job/ci_linux-centos/2/
Example packaging job: https://ci.ros2.org/job/ci_packaging_linux-centos/15/

**Comment by cottsay** [comment]
Reminder that the node creation needs to be updated to include the `linux_centos` tag before we merge this. I did it manually for testing.

**Comment by dirk-thomas** [comment]
The Cent OS job should be excluded from the launcher job.

Other than that it looks good to me.

**Comment by cottsay** [comment]
> The Cent OS job should be excluded from the launcher job.

I agree, and I made the change. Thanks!

**Comment by cottsay** [comment]
I'm going to hold off merging this until we can update the AWS images with the new label. Hopefully @nuclearsandwich and I can work that out tomorrow.

File: create_jenkins_job.py [hunk 1]
```diff
@@ -114,6 +114,25 @@ def main(argv=None):
             'shell_type': 'Shell',
             'ignore_rmw_default': data['ignore_rmw_default'] | {'rmw_connext_cpp', 'rmw_connext_dynamic_cpp'},
         },
+        'linux-centos': {
+            'label_expression': 'linux_centos',
```
Comment (nuclearsandwich) [inline create_jenkins_job.py]: As @cottsay and I discussed offline, there's no special capability or change needed on our existing linux agents to run centos containers. These can just use the regular linux nodes and thus no infrastructure changes are required.

**Comment by dirk-thomas** [comment]
The repeated nightly fails because it doesn't have the `--packages-skip*` arguments it is supposed to have. Please fix the logic to make sure the test arguments are as intended.

**Pull request merged**